pkg/api/microservices/microservices_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package microservices
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-crd-generator/pkg/apis/verrazzano/v1beta1"
+	"github.com/verrazzano/verrazzano-operator/pkg/controller"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutilcontroller"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestReturnAllMicroservices tests that all Microservices are returned
+// GIVEN that Microservices are configured
+// WHEN I call GetAllMicroservices
+// THEN all the Microservices should be returned
+// AND that the Microservices returned should be the correct amount and have valid values
+func TestReturnAllMicroservices(t *testing.T) {
+	var modelBindingPairs = map[string]*types.ModelBindingPair{
+		"test-pair-1": testutil.GetModelBindingPair(),
+	}
+	// GIVEN empty model binding pairs, managed clusters and fake kubernetes clients.
+	var clients kubernetes.Interface = k8sfake.NewSimpleClientset()
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+
+	type args struct {
+		w http.ResponseWriter
+		r *http.Request
+	}
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{
+			name:    "verifyReturnAllMicroservices",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var myMicroservices []Microservice
+			Init(tt.listers)
+			resp := httptest.NewRecorder()
+			ReturnAllMicroservices(resp, httptest.NewRequest("GET", "/microservices", nil))
+			assert.Equal(t, http.StatusOK, resp.Code, "expect the http return code to be http.StatusOk")
```

Comment:
do the following so you don't have to pass t to every assert
	assert := assert.New(t)
        assert.Equal(http.StatusOK, ...)

Reply:
Done